josh-json/src/files.js

```diff
@@ -0,0 +1,124 @@
+const atomic = require('atomically');
+const path = require('path');
+const { serializeData, parseData } = require('./utils');
+const fs = require('fs');
+const { isEqual } = require('lodash');
+class FileManager {
+  constructor(dir, opts) {
+    this.options = opts || {};
+    this.dir = path.resolve(process.cwd(), dir);
+    try {
+      fs.mkdirSync(dir);
+    } catch {
+      // don't need to do anything
+    }
+  }
+  async getIndex(key) {
+    let index = await this.getFile('index.json');
```

Comment:
Would you not be able to read directly from the folder with readDir or something? Or is this more performant overall?

Reply:
Yea so having an index seemed a lot more performant on where keys were instead of having to go through each file

Reply:
Essentially the structure is 
index.json:
```json
{ 
   files: [
      {
         location: "resolved/path/to/file"
         keys: ["hello", "foo", "test"] // up to options.maxLength which defaults to 100 at which it makes a new file
      }
   ]
}
```